src/include/timestamp.h

```diff
@@ -0,0 +1,52 @@
+/*-
+ * Copyright (c) 2014-2020 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * We format timestamps in a couple of ways, declare appropriate sized buffers. Hexadecimal is 2x
+ * the size of the value. MongoDB format (high/low pairs of 4B unsigned integers, with surrounding
+ * parenthesis and separating comma and space), is 2x the maximum digits from a 4B unsigned integer
+ * plus 4. Both sizes include a trailing null byte as well.
+ */
+#define WT_TS_HEX_STRING_SIZE (2 * sizeof(wt_timestamp_t) + 1)
+#define WT_TS_INT_STRING_SIZE (2 * 10 + 4 + 1)
+
+/*
+ * We need an appropriately sized buffer for formatted time pairs, aggregates and windows. This is
+ * for time windows with 4 timestamps, 2 transaction IDs, prepare state and formatting. The
+ * formatting is currently about 32 characters - enough space that we don't need to think about it.
+ */
+#define WT_TP_STRING_SIZE (WT_TS_INT_STRING_SIZE + 1 + 20 + 1)
```

Comment:
Do we still use time pairs? I feel we can remove this and WT_TIME_PAIR.

Reply:
It's still used in a meaningful way by `__sync_ref_obsolete_check` - so I don't think we can get rid of it. I've cleaned up a misuse in `bt_vrfy.c`.

Reply:
Yes, WT_TP_STRING_SIZE is still used. I just feel a bit weird that we have both WT_TIME_WINDOW struct and WT_TIME_PAIR struct. Should we either make WT_TIME_PAIR as fields in time window or just remove WT_TIME_PAIR? This is another big change so I don't feel we should spend time on that now. I'm happy to merge this first and decide what to do with WT_TIME_PAIR later.

Reply:
I think we should rename WT_CELL_UNPACK_TIME_PAIRS_CLEARED to WT_CELL_UNPACK_TIME_WINDOW_CLEARED

Reply:
I agree with @quchenhao (nice catch!).

I renamed `WT_TIME_PAIR` to `WT_HS_TIME_PAIR` and made it local to `history/hs.c`, the only remaining user.

I renamed `WT_CELL_UNPACK_TIME_PAIRS_CLEARED` to `WT_CELL_UNPACK_TIME_WINDOW_CLEARED`.

I left `WT_TP_STRING_SIZE` and `__wt_time_pair_to_string` alone, I agree with @agorrod on that one.